packages/amplify_api/test/amplify_rest_api_methods_test.dart

```diff
@@ -0,0 +1,212 @@
+    var responseData = Uint8List.fromList(
+        "{\"success\":\"get call succeed!\",\"url\":\"/items\"}".codeUnits);
+
+    apiChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "get") {
+        Map<dynamic, dynamic> restOptions = methodCall.arguments["restOptions"];
+        expect(restOptions["path"], "/items");
+
+        return {"data": responseData};
+      }
+    });
+
+    RestOperation restOperation = api.get(
+        restOptions: RestOptions(
+      path: "/items",
+    ));
+
+    RestResponse response = await restOperation.response;
+
+    expect(response.data, responseData);
+  });
+
+  test('DELETE returns proper response.data', () async {
+    var responseData = Uint8List.fromList(
+        "{\"success\":\"delete call succeed!\",\"url\":\"/items\"}".codeUnits);
+
+    apiChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "delete") {
+        Map<dynamic, dynamic> restOptions = methodCall.arguments["restOptions"];
+        expect(restOptions["path"], "/items");
+
+        return {"data": responseData};
+      }
+    });
+
+    RestOperation restOperation = api.delete(
+        restOptions: RestOptions(
+      path: "/items",
+    ));
+
+    RestResponse response = await restOperation.response;
+
+    expect(response.data, responseData);
+  });
+
+  test('GET Status Code Error throws proper error', () async {
+    var errorResponseMap = {
+      "PLATFORM_EXCEPTIONS": {
+        "platform": "iOS",
+        "localizedErrorMessage": "The HTTP response status code is [400].",
+        "recoverySuggestion":
+            "The metadata associated with the response is contained in the HTTPURLResponse.\nFor more information on HTTP status codes, take a look at\nhttps://en.wikipedia.org/wiki/List_of_HTTP_status_codes"
+      }
+    };
+
+    apiChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "get") {
+        throw PlatformException(
+            code: "AmplifyException",
+            message: "AMPLIFY_API_GET_FAILED",
+            details: errorResponseMap);
+      }
+    });
+
+    try {
+      RestOperation restOperation = api.get(
+          restOptions: RestOptions(
+        path: "/items",
+      ));
+      await restOperation.response;
+    } on ApiError catch (e) {
+      expect(e.code, "AmplifyException");
+      expect(e.message, "AMPLIFY_API_GET_FAILED");
+
+      Map<dynamic, dynamic> platformExceptions =
+          (e.details)["PLATFORM_EXCEPTIONS"];
+
+      expect(platformExceptions["platform"], "iOS");
+      expect(platformExceptions["localizedErrorMessage"],
+          "The HTTP response status code is [400].");
+      expect(platformExceptions["recoverySuggestion"],
+          "The metadata associated with the response is contained in the HTTPURLResponse.\nFor more information on HTTP status codes, take a look at\nhttps://en.wikipedia.org/wiki/List_of_HTTP_status_codes");
```

Comment:
Move to constant?

Reply:
Switched to using a referenceError comparison :

```
  let referenceError = APIError.httpStatusError(400, HTTPURLResponse())
                
                XCTAssertEqual(errorMap["PLATFORM_EXCEPTIONS"]!["platform"], "iOS")
                XCTAssertEqual(errorMap["PLATFORM_EXCEPTIONS"]!["localizedErrorMessage"], referenceError.errorDescription)
                XCTAssertEqual(errorMap["PLATFORM_EXCEPTIONS"]!["recoverySuggestion"], referenceError.recoverySuggestion)
```